worlds/hk/__init__.py

```diff
@@ -218,11 +247,58 @@ def create_items(self):
             for loc, price in zip(locations, prices):
                 loc.cost = price
 
+    def _can_beat_thk(self, state, player):
+        return (
+            state.has('Opened_Black_Egg_Temple', player)
+            and (state.count('FIREBALL', player) + state.count('SCREAM', player) + state.count('QUAKE', player)) > 1
+            and (  # NAILCOMBAT
+                    state.has('LEFTSLASH', player)
+                    or state.has('RIGHTSLASH', player)
+                    or state.has('UPSLASH', player)
+            )
+            and (
+                    (state.has('LEFTDASH', player) or state.has('RIGHTDASH', player))
+                    or self.world.ProficientCombat[self.player]
+            )
+        )
+
     def set_rules(self):
         world = self.world
         player = self.player
         if world.logic[player] != 'nologic':
-            world.completion_condition[player] = lambda state: state.has('DREAMER', player, 3)
+            goal = world.Goal[player]
+            if goal == Goal.option_siblings:
+                world.completion_condition[player] = lambda state: (
+                    self._can_beat_thk(state, player)
+                    and state.has('WHITEFRAGMENT', player, 3)
+                )
+            elif goal == Goal.option_radiance:
+                world.completion_condition[player] = lambda state: (
+                    state.count('Opened_Black_Egg_Temple', player)
+                    and state.has('WHITEFRAGMENT', player, 3)
+                    and state.has('DREAMNAIL', player, 1)
+                    and (
+                        (state.has('LEFTCLAW', player) and state.has('RIGHTCLAW', player))
+                        or state.has('WINGS', player)
+                    )
+                    and (
+                        state.count('FIREBALL', player) + state.count('SCREAM', player)
+                        + state.count('QUAKE', player)
+                    ) > 1
+                    and (  # NAILCOMBAT
+                        state.count('LEFTSLASH', player)
+                        or state.count('RIGHTSLASH', player)
+                        or state.count('UPSLASH', player)
+                    )
+                    and (
+                        (state.has('LEFTDASH', player, 2) and state.has('RIGHTDASH', player, 2))  # Both Shade Cloaks
+                        or (self.world.ProficientCombat[self.player] and state.has('QUAKE', player))  # or Dive
+                    )
+                )
```

Comment:
Whatever logic calculations need to happen here should be in `HKLogicMixin` if possible then referred to here.

Reply:
This should now be done.  HKLogicMixin didn't exist when I first opened this, hopefully I used it correctly.